pkg/opensearch/response_parser_test.go

```diff
@@ -2659,3 +2659,249 @@ func TestProcessTraceListResponse(t *testing.T) {
 }
+
+
+func TestProcessSpansResponse_withMultipleSpansQueries(t *testing.T) {
+	targets := map[string]string{
+		"A": `{
+			"timeField": "@timestamp",
+			"metrics": [{ "type": "count", "id": "1" }],
+			"query": "traceId:test",
+			"luceneQueryType": "Traces"
+			}`,
+		"B": `{
+			"timeField": "@timestamp",
+			"metrics": [{ "type": "count", "id": "1" }],
+			"query": "traceId:test123",
+			"luceneQueryType": "Traces"
+			}`,
+	}
+
+	response := `
+	{
+		"responses": [
+			{
+				"hits": {
+					"hits": [
+						{
+							"_source": {
+								"traceId": "000000000000000047ed3a25a7dba0cd",
+								"droppedLinksCount": 0,
+								"kind": "SPAN_KIND_SERVER",
+								"droppedEventsCount": 0,
+								"traceGroupFields": {
+									"endTime": "2023-10-18T07:58:38.689468Z",
+									"durationInNanos": 870879000,
+									"statusCode": 0
+								},
+								"traceGroup": "HTTP GET /dispatch",
+								"serviceName": "frontend",
+								"parentSpanId": "",
+								"spanId": "47ed3a25a7dba0cd",
+								"traceState": "",
+								"name": "HTTP GET /dispatch",
+								"startTime": "2023-10-18T07:58:37.818589Z",
+								"links": [],
+								"endTime": "2023-10-18T07:58:38.689468Z",
+								"droppedAttributesCount": 0,
+								"durationInNanos": 870879000,
+								"events": [],
+								"span.attributes.sampler@param": true,
+								"span.attributes.http@method": "GET",
+								"resource.attributes.client-uuid": "1ba5d5eb37e7c2a1",
+								"status.code": 0
+							}
+						}
+						]
+					}
+				},
+				{
+					"hits": {
+						"hits": [
+							{
+								"_source": {
+									"traceId": "000000000000000047ed3a25a7dba0cd",
+									"droppedLinksCount": 0,
+									"kind": "SPAN_KIND_SERVER",
+									"droppedEventsCount": 0,
+									"traceGroupFields": {
+										"endTime": "2023-10-18T07:58:38.689468Z",
+										"durationInNanos": 870879000,
+										"statusCode": 0
+									},
+									"traceGroup": "HTTP GET /dispatch",
+									"serviceName": "frontend",
+									"parentSpanId": "",
+									"spanId": "47ed3a25a7dba0cd",
+									"traceState": "",
+									"name": "HTTP GET /dispatch",
+									"startTime": "2023-10-18T07:58:37.818589Z",
+									"links": [],
+									"endTime": "2023-10-18T07:58:38.689468Z",
+									"droppedAttributesCount": 0,
+									"durationInNanos": 870879000,
+									"events": [],
+									"span.attributes.sampler@param": true,
+									"resource.attributes.client-uuid": "1ba5d5eb37e7c2a1",
+									"status.code": 0
+								}
+							}
+							]
+						}
+					}
+			
+	]	
+	}
+	`
+
+	rp, err := newResponseParserForTest(targets, response, nil, client.ConfiguredFields{TimeField: "@timestamp"}, &backend.DataSourceInstanceSettings{UID: "123", Name: "DatasourceInstanceName"})
+	assert.Nil(t, err)
+
+	result, err := rp.parseResponse()
+	require.NoError(t, err)
+	require.Len(t, result.Responses, 2)
+
+	// 1st query
+	queryResTraceSpans1 := result.Responses["B"]
```

Comment:
Should one of these be `A`? Also totally optional, I know the original was very bare bones, but is there anything else we want to check in these to confirm parsing?

Reply:
Done, added a check for operationName :)